src/test/ts/Exercise3OptionTest.ts

```diff
@@ -12,9 +23,76 @@ UnitTest.test('getProtocol', () => {
   Assert.eq('malformed protocol', Optional.none<string>(), Ex.getProtocol('3ttp://frog.com'), tOptional());
 });
 
-UnitTest.test('toPositiveInteger', () => {
-  // TODO: write a few test cases
-  // NOTE the examples above - due to how Optional is implemented, we can't compare values as JS objects/arrays/values,
-  // and we need to pass in the tOption() equality value
+UnitTest.test('getNextSibling', () => {
+  const div = document.createElement('div');
+  Assert.eq('should return none', Optional.none<ChildNode>(), Ex.getNextSibling(div), tOptional());
+  
+  const parent = document.createElement('div');
+  const child1 = document.createElement('span');
+  const child2 = document.createElement('span');
+  parent.appendChild(child1);
+  parent.appendChild(child2);
+  Assert.eq('should return some', Optional.some(parent.childNodes.item(1)), Ex.getNextSibling(child1), tOptional());
+});
+
+UnitTest.test('getAttributeNode', () => {
+  const div = document.createElement('div');
+  Assert.eq('should return none', Optional.none<Attr>(), Ex.getAttributeNode(div, 'id'), tOptional());
+
+  const divWithId = document.createElement('div');
+  divWithId.setAttribute('id', 'foo');
+  Assert.eq('should return some', Optional.some<any>(divWithId.getAttributeNode('id')), Ex.getAttributeNode(divWithId, 'id'), tOptional());
```

Comment:
Generally, if you leave the `any` out typescript should figure out
```suggestion
  Assert.eq('should return some', Optional.some(divWithId.getAttributeNode('id')), Ex.getAttributeNode(divWithId, 'id'), tOptional());
```

However, if you get compiler errors from that, the correct type should be `Attr` apparently